### PR TITLE
[CC-1022] Updates

### PR DESCRIPTION
**Pull request by mafrosis**
Probably the only two contentious commits are:

 - 1861609 - renaming `application.py` to `manage.py`. This is primarily to remove ambiguity and (as I'm sure you know) the `manage.py` pattern is well know from the Django community.
 - 504f700 - renaming `app.py` to `config.py`, and then using `app.py` purely as the WSGI entrypoint. This a separation of concerns thing, and works well as a codebase scales.

https://artsalliancemedia.atlassian.net/browse/CC-1022

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,17 +47,17 @@ dependencies-dev: dependencies-python
 	$(PIP) install -r env_conf/requirements-dev.txt
 
 lint:
-	$(FLAKE8) --config=.flake8rc application
+	$(FLAKE8) --config=.flake8rc $(PROJECT)
```
Comment (RangerOfFire) [inline Makefile]: Shouldn't this still be application as the folder in the repo root is still called application?

File: application/config.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import os
+from logging import StreamHandler
+
+from flask import Flask
+
+from application.env_utils import load_config
```
Comment (RangerOfFire) [inline application/config.py]: It feels like this file has very little to do with config, if this is an accepted design pattern then ignore me.

File: env_conf/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM python:2.7-slim
+FROM python:3.4-slim
```
Comment (RangerOfFire) [inline env_conf/Dockerfile]: :+1:

File: test/conftest.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    WHERE pg_stat_activity.datname = '{}'
+      AND pid <> pg_backend_pid();
+    '''.format(db_name)
+    engine.execute(text)
+
+
+@pytest.yield_fixture(scope='session')
+def flask_app():
+    """
+    Provide test isolation by creating
```
Comment (RangerOfFire) [inline test/conftest.py]: This docstring feels unfinished :racehorse:

**Comment by RangerOfFire** [comment]
Structure looks good, just the minor things I commented on.

File: test/conftest.py (hunk 4 above)
Comment (mafrosis) [inline test/conftest.py]: Good spot

**Comment by shipperizer** [comment]
looks good, I would just copy the uwsgi config as well and maybe the logging part from the user-service, but it's arguable, so up to you
here's my :sweet_potato:

File: Makefile (hunk 1 above)
Comment (mafrosis) [inline Makefile]: The project var is defined at the top of the Makefile

File: application/config.py (hunk 2 above)
Comment (mafrosis) [inline application/config.py]: I had meant to merge `env_utils.py` into `config.py`.. Not sure how I missed that :) Update coming.

**Comment by mafrosis** [comment]
@shipperizer I thought we could add the logging stuff when we've finalised it. Maybe it's ready now?

File: Makefile (hunk 1 above)
Comment (mafrosis) [inline Makefile]: You are absolutely right, I was being damned fool

**Comment by shipperizer** [comment]
@mafrosis good to me, i was just lazy and didn't want to do it on my own :dart:

**Comment by mafrosis** [comment]
@shipperizer @RangerOfFire 
Notable changes in recent iteration:
- renamed the `application` app to `flask_template`, to make it explicit and unambiguous
- merged `env_utils.py` and `config.py`, and removed unnecessarily complicated `load_env_var()` function
- included logging code and uwsgi config :)

**Comment by shipperizer** [comment]
:sweet_potato:

**Comment by lorantonodi** [comment]
👍

File: flask_template/config.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    return app
+
+
+def load_config():
+    config = {}
+
+    # load DEBUG env var and convert to boolean
+    config['DEBUG'] = os.environ.get('DEBUG', False)
+    if isinstance(config['DEBUG'], str) and config['DEBUG'].lower() == 'true':
```
Comment (RangerOfFire) [inline flask_template/config.py]: I'm slightly iffy about the need for this but whatever it works.

**Comment by RangerOfFire** [comment]
:accept:

File: flask_template/config.py (hunk 5 above)
Comment (mafrosis) [inline flask_template/config.py]: The previous implemented of this created a `value_is` lambda which is passed into `load_env_var` where it wraps `os.environ.get`:

https://github.com/artsalliancemedia/flask-template/blob/master/application/env_utils.py#L28

I thought that was a bit over-engineered. Hence the very simple alternative approach.

**Pull request merged**